.github/workflows/firmware.yml

```diff
@@ -5,13 +5,32 @@ on:
   push:
     tags:
       - v*
+  pull_request:
 
 jobs:
+  generate_target_matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      target_json: ${{ steps.set_target.outputs.target }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
+    - name: Set target matrix
+      id: set_target
+      shell: bash
+      run: |
+        if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
+          target_list="[\"ipq40xx-generic\", \"ramips-mt76x8\", \"x86-64\", \"x86-generic\"]"
+        else
+          target_list="[\"ath79-generic\", \"ath79-nand\", \"bcm27xx-bcm2708\", \"bcm27xx-bcm2709\", \"bcm27xx-bcm2710\", \"ipq40xx-generic\", \"ipq806x-generic\", \"lantiq-xway\", \"lantiq-xrx200\", \"mediatek-mt7622\", \"mpc85xx-p1010\", \"mpc85xx-p1020\", \"mvebu-cortexa9\", \"ramips-mt7620\", \"ramips-mt7621\", \"ramips-mt76x8\", \"ramips-rt305x\", \"rockchip-armv8\", \"sunxi-cortexa7\", \"x86-64\", \"x86-generic\", \"x86-geode\", \"x86-legacy\"]"
+        fi
+        echo ::set-output name=target::{\"target\": $(echo $target_list)}\"
```

Comment:
Wenn hier sowieso eine Shell laufen muss, warum dann nicht die Target Liste gleich dynamisch generieren lassen?

Das wäre z.B. das Bash Konstrukt welches wir einsetzen:

```
GLUON_TARGETS=$(make list-targets | grep -Ev '(mvebu-cortexa9|x86-legacy|mpc85xx-generic|mpc85xx-p1010|ramips-rt305x)' | tr '\n' ' ')
```
Das erzeugt eine Liste mit allem außer den gennanten.

Reply:
Ich finde eine statische Liste sinnvoll und verständlich. Ich würde nur gerne etwas hinzufügen wollen für einen bestimmten PR, wenn ich einen Fix mache für ein bestimmtes Device, dann will ich das im PR-Bulid auch gebaut bekommen.

Reply:
Diese statische Liste funktioniert muss dann aber z.B. für OpenWRT 21.02 anders sein als 19.07 usw.

Ich weiß gerade nicht wie das gelöst ist, aber es gibt schon mal neue Targets und alte fallen weg. Die sorgen darüber kann man sich einfach ersparen wenn man es automatisch macht.

Reply:
Please work with static lists, otherwise new targets are added when a new Gluon version is pushed and that's maybe on a first build not what we want.

Reply:
Changed it